include/logging/log_core.h

```diff
@@ -283,12 +320,51 @@ static inline char z_log_minimal_level_to_char(int level)
 		}							       \
 	} while (false)
 
+
+#define __LOG_VA(_level, _id, _filter, _format, _valist, _argnum, _strdup) \
```

Comment:
this also seems like a function converted to macro. I don't see it making any use of preprocessor.

Reply:
it is using c to ensure that log macros are compiled and do not generate warning (unused variables) when log level is disabled.
If it was a macro then code below would generate warning that `ms` is unused when logger is off:
```
u32_t ms = conv_to_ms(ticks);
LOG_INF("tick:%d, ms:%d", ticks, ms);
```